ksql-version-metrics-client/src/main/java/io/confluent/ksql/version/metrics/KsqlVersionCheckerAgent.java

```diff
@@ -19,48 +19,70 @@
 import io.confluent.ksql.version.metrics.collector.KsqlModuleType;
 import io.confluent.support.metrics.BaseSupportConfig;
 import io.confluent.support.metrics.PhoneHomeConfig;
+import java.time.Clock;
+import java.util.Objects;
+import java.util.Optional;
 import java.util.Properties;
+import java.util.concurrent.TimeUnit;
+import java.util.function.Supplier;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 public class KsqlVersionCheckerAgent implements VersionCheckerAgent {
 
-  private KsqlVersionChecker ksqlVersionChecker;
+  private static final long MAX_INTERVAL = TimeUnit.DAYS.toMillis(1);
+
 
   private boolean enableSettlingTime;
 
+  private Clock clock;
+
+  private volatile long requestTime;
+  private final Supplier<Boolean> activeQuerySupplier;
+
   private static final Logger log = LoggerFactory.getLogger(KsqlVersionCheckerAgent.class);
 
-  public KsqlVersionCheckerAgent() {
-    this(true);
+  public KsqlVersionCheckerAgent(final Supplier<Boolean> activeQuerySupplier) {
```

Comment:
A java doc wouldn't go a miss:

```suggestion
  /**
   * @param activeQuerySupplier supplier that indicates if there are any active persistent queries
   */
  public KsqlVersionCheckerAgent(final Supplier<Boolean> activeQuerySupplier) {
```